src/link_prober/LinkProber.cpp

```diff
@@ -313,26 +308,42 @@ void LinkProber::handleRecv(
                 mMuxPortConfig.getPortName() %
                 mMuxPortConfig.getBladeIpv4Address().to_string()
             );
-            if (memcmp(icmpPayload->un.uuid.data, IcmpPayload::getGuidData(),
-                       IcmpPayload::getGuid().size()) == 0) {
+            bool isMatch = memcmp(icmpPayload->uuid, IcmpPayload::getGuidData(), sizeof(icmpPayload->uuid));
+            if (isMatch)
```

Comment:
If icmpPayload->uuid == icmpPayload->uuid, isMatch will be zero.
Please double confirm if the logic is expected.